<launch>


<!-- name: ['left_e0', 'left_e1', 'left_s0', 'left_s1', 'left_w0', 'left_w1', 'left_w2']
position: [-0.5852136705782853, 1.492946801809479, -0.16835439147042416, -0.7071651432151758, 0.5641214347448609, 0.9491506125040997, -0.19711653124327566] -->

  <!-- Good position for the arm so the camera can see all 3 objects -->
  <param name="left_w0" value="0.5641214347448609"/>
  <param name="left_w1" value="0.9491506125040997"/>
  <param name="left_w2" value="-0.19711653124327566"/>
  <param name="left_e0" value="-0.5852136705782853"/>
  <param name="left_e1" value="1.492946801809479"/>
  <param name="left_s0" value="-0.16835439147042416"/>
  <param name="left_s1" value="-0.7071651432151758"/>

  <node pkg="tf" type="static_transform_publisher" name="link1_broadcaster" args="1.28 0.10 0.19 0 0 0 base camera_link 1000" />

  <include file="$(find baxter_moveit_config)/launch/planning_context.launch">
    <arg name="load_robot_description" value="true"/>
  </include>

  <include file="$(find baxter_moveit_config)/launch/move_group.launch">
    <arg name="allow_trajectory_execution" value="true"/>
  </include>

  <!-- Specifies which camera to use and the resolution -->
  <node name="camera_control" pkg="baxter_tools" type="camera_control.py"
    args="-o left_hand_camera -r 640x400"/>

  <!-- Gripper Argument -->
  <!-- might need to change this to left -->
  <arg name="gripper" default="left" />

  <!-- Start the Gripper Action Server -->
  <node name="rsdk_gripper_action_server" pkg="baxter_interface" type="gripper_action_server.py" args="--gripper $(arg gripper)" required="true" />

  <node pkg="baxter_interface" type="joint_trajectory_action_server.py" name="joint_traj_action_srv" output="screen" respawn="false" />

  <!-- up to date -->
  <node pkg="object_recognition" type="baxter_object_prediction.py" name="baxter_object_prediction" output="screen" respawn="true"/>

  <!-- need to modify this somehow -->
  <node pkg="object_recognition" type="pickup_object.py" name="pickup_object" output="screen" respawn="false" launch-prefix="xterm -e "/>

</launch>
